hudi-io/src/main/java/org/apache/hudi/storage/StoragePathInfo.java

```diff
@@ -109,6 +109,11 @@ public long getModificationTime() {
     return modificationTime;
   }
 
+  @Override
+  public int compareTo(StoragePathInfo o) {
+    return this.getPath().compareTo(o.getPath());
```

Comment:
Practically, would it ever hit a scenario when this.getPath() == o.getPath()? If so, do need to break ties? If not, should we add a validation and throw IllegalStateException?

Reply:
Also, is there a need for null check?

Reply:
There is no need for null check; path should always exist.

We shouldn’t have a case for multiple path info instances with the same path, in the same file listing call.  So this should be OK.  And this is only used in Hudi CLI for sorting purpose.